### PR TITLE
Wire up support for converting integrations-next configs to flow.

### PR DESCRIPTION
**Pull request by erikbaranowski**
Include a mechanism for passing command line flags to the converter from the old format.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: converter/internal/staticconvert/staticconvert.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func Convert(in []byte, extraArgs []string) ([]byte, diag.Diagnostics) {
 	var diags diag.Diagnostics
 
 	fs := flag.NewFlagSet("convert", flag.ContinueOnError)
-	args := []string{"-config.file", "convert", "-config.expand-env"}
+	args := []string{"-config.file", "convert"}
```
Comment (erikbaranowski) [inline converter/internal/staticconvert/staticconvert.go:34]: Remove this hard coded since we can now pass this in and have more control over the converter. This was the lesser of 2 evils before to automatically expand environment vars.

File: converter/internal/staticconvert/testdata-v2/integrations_v2.yaml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+server:
+  log_level: ${SOME_ENVIRONMENT_VARIABLE:='debug'}
```
Comment (erikbaranowski) [inline converter/internal/staticconvert/testdata-v2/integrations_v2.yaml:2]: Added an integration-next test for env vars to validate it also works.

File: converter/internal/staticconvert/staticconvert_test.go [hunk 3]
```diff
@@ -10,8 +10,8 @@ import (
 )
 
 func TestConvert(t *testing.T) {
-	test_common.TestDirectory(t, "testdata", ".yaml", true, []string{}, staticconvert.Convert)
-	test_common.TestDirectory(t, "testdata-v2", ".yaml", true, []string{"-enable-features", "integrations-next"}, staticconvert.Convert)
+	test_common.TestDirectory(t, "testdata", ".yaml", true, []string{"-config.expand-env"}, staticconvert.Convert)
```
Comment (erikbaranowski) [inline converter/internal/staticconvert/staticconvert_test.go:13]: This is no longer hard coded to be added so we need to pass it in for the tests.

File: cmd/internal/flowmode/cmd_convert.go [hunk 4]
```diff
@@ -41,7 +42,10 @@ The -f flag can be used to specify the format we are converting from.
 
 The -b flag can be used to bypass errors. Errors are defined as 
 non-critical issues identified during the conversion where an
-output can still be generated.`,
+output can still be generated.
+
+The -e flag can be used to pass extra arguments to the converter
+which used by the original format.`,
```
Comment (thampiotr) [inline cmd/internal/flowmode/cmd_convert.go:48]:
```suggestion
which were used by the original format.`,
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 5]
```diff
@@ -305,7 +331,7 @@ After the configuration is converted, review the {{< param "PRODUCT_NAME" >}} co
 
 Review the following checklist:
 
-* The following configuration options are not available for conversion to {{< param "PRODUCT_NAME" >}}: [Integrations next][], [Traces][], and [Agent Management][].
+* The following configuration options are not available for conversion to {{< param "PRODUCT_NAME" >}}: [Traces][] and [Agent Management][].
```
Comment (thampiotr) [inline docs/sources/flow/getting-started/migrating-from-static.md:334]: Maybe a bit of a scope creep, but should we clarify that traces themselves are supported? It was one of the recent questions we got.

File: cmd/internal/flowmode/cmd_run.go [hunk 6]
```diff
@@ -123,6 +123,7 @@ depending on the nature of the reload error.
 		BoolVar(&r.disableReporting, "disable-reporting", r.disableReporting, "Disable reporting of enabled components to Grafana.")
 	cmd.Flags().StringVar(&r.configFormat, "config.format", r.configFormat, fmt.Sprintf("The format of the source file. Supported formats: %s.", supportedFormatsList()))
 	cmd.Flags().BoolVar(&r.configBypassConversionErrors, "config.bypass-conversion-errors", r.configBypassConversionErrors, "Enable bypassing errors when converting")
+	cmd.Flags().StringVar(&r.configExtraArgs, "config.extra-args", r.configExtraArgs, "Extra arguments from the original format used by the converter")
```
Comment (thampiotr) [inline cmd/internal/flowmode/cmd_run.go:126]: Nit: maybe an example or a mention that they can be separated by spaces?
For the people that don't read docs, just read the help messages.

File: cmd/internal/flowmode/cmd_convert.go [hunk 7]
```diff
@@ -174,3 +180,15 @@ func supportedFormatsList() string {
...
 }
+
+func parseExtraArgs(extraArgs string) []string {
+	var result []string
+	if extraArgs != "" {
+		arguments := strings.Fields(extraArgs)
+		for _, arg := range arguments {
+			parts := strings.Split(arg, "=")
+			result = append(result, parts...)
+		}
```
Comment (thampiotr) [inline cmd/internal/flowmode/cmd_convert.go:191]: Hmm, what if someone passes their extra args using another (valid in Cobra library we use) syntax? I think the following are all valid ways: `--key=value`, `--key value`, `-k=value` (when there is shorthand defined), `-k value`, `--boolVariable` (makes the boolVariable True). 

I wonder if we can use Cobra library to somehow do the parsing instead?
Also a test for this function would be a good idea.

File: converter/internal/staticconvert/testdata-v2/integrations_v2.river [hunk 8]
```diff
@@ -16,6 +16,11 @@ loki.write "logs_log_config" {
 	external_labels = {}
 }
 
+logging {
```
Comment (mattdurham) [inline converter/internal/staticconvert/testdata-v2/integrations_v2.river:19]: Where did this change come from? Level debug feels odd for a default.

Comment (mattdurham) [reply converter/internal/staticconvert/testdata-v2/integrations_v2.river:19]: NVM literally see if below

**Review by mattdurham — APPROVED** [review]
LGTM

File: docs/sources/flow/getting-started/migrating-from-static.md (hunk 5 above)
Comment (erikbaranowski) [reply docs/sources/flow/getting-started/migrating-from-static.md:334]: added some wording, LMK what you think

File: cmd/internal/flowmode/cmd_run.go (hunk 6 above)
Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_run.go]: added

File: cmd/internal/flowmode/cmd_convert.go (hunk 7 above)
Comment (thampiotr) [reply cmd/internal/flowmode/cmd_convert.go:192]: Took a closer look at this and we don't use Cobra in static mode. Also I think this should handle most cases other than trying to pass `-something="key=value,key2=value2"`, but perhaps that's too much of an edge case?

Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_convert.go:192]: It's a good point though to look closer and add some tests so it's clear how this works and what it doesn't work with. I can spend some time trying to clean up the implementation on this one since it is a bit brittle as implemented.

File: docs/sources/flow/getting-started/migrating-from-prometheus.md [hunk 9]
```diff
@@ -225,7 +225,7 @@ Configuration conversion is done on a best-effort basis. {{< param "PRODUCT_ROOT
 
 After the configuration is converted, review the {{< param "PRODUCT_NAME" >}} configuration file created and verify that it's correct before starting to use it in a production environment.
 
-Review the following checklist:
+This following list of limitations is specific to the converter tooling and not {{< param "PRODUCT_NAME" >}}:
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-prometheus.md:228]:
```suggestion
The following list is specific to the convert command and not {{< param "PRODUCT_NAME" >}}:
```

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 10]
```diff
@@ -204,7 +204,7 @@ Configuration conversion is done on a best-effort basis. {{< param "PRODUCT_ROOT
 
 After the configuration is converted, review the {{< param "PRODUCT_NAME" >}} configuration file created and verify that it's correct before starting to use it in a production environment.
 
-Review the following checklist:
+This following list of limitations is specific to the converter tooling and not {{< param "PRODUCT_NAME" >}}:
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:207]:
```suggestion
The following list is specific to the convert command and not {{< param "PRODUCT_NAME" >}}:
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 11]
```diff
@@ -297,15 +297,41 @@ loki.write "logs_varlogs" {
 }
 ```
 
+## Integrations Next
+
+[Integrations next][] configs can be converter or run by adding the `extra-args` flag for [convert][] or `config.extra-args` for [run][].
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-static.md:302]:
```suggestion
You can convert the [integrations next][] configurations, or you can add the `extra-args` flag for [convert][] or `config.extra-args` for [run][] to use the configurations as-is.
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 12]
```diff
@@ -297,15 +297,41 @@ loki.write "logs_varlogs" {
...
+```static-binary
+AGENT_MODE=flow grafana-agent convert --source-format=static --extra-args="-enable-features=integrations-next" --output=<OUTPUT_CONFIG_PATH> <INPUT_CONFIG_PATH>
+```
+
+```flow-binary
+grafana-agent-flow convert --source-format=static --extra-args="-enable-features=integrations-next" --output=<OUTPUT_CONFIG_PATH> <INPUT_CONFIG_PATH>
+```
+
+{{< /code >}}
+
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-static.md:315]:
```suggestion

 Replace the following:
   * _`<INPUT_CONFIG_PATH>`_: The full path to the [Static][] configuration.
   * _`<OUTPUT_CONFIG_PATH>`_: The full path to output the {{< param "PRODUCT_NAME" >}} configuration.
   
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 13]
```diff
@@ -297,15 +297,41 @@ loki.write "logs_varlogs" {
...
+## Environment Vars
+
+You can interpret env variables in your static config using the
+`-config.expand-env` command line flag. This can be passed to [convert][] by
+including `--extra-args="-config.expand-env"` or to [run][]
+by including `--config.extra-args="-config.expand-env"`.
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-static.md:321]:
```suggestion
You can use the `-config.expand-env` command line flag to interpret environment variables in your Grafana Agent Static configuration.
You can pass these flags to [convert][] with `--extra-args="-config.expand-env"` or to [run][] with `--config.extra-args="-config.expand-env"`.
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 14]
```diff
@@ -297,15 +297,41 @@ loki.write "logs_varlogs" {
...
+
+> It is possible to combine `integrations-next` with `expand-env`.
+> For [convert][], `--extra-args="-enable-features=integrations-next -config.expand-env"`
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-static.md:324]:
```suggestion
> It's possible to combine `integrations-next` with `expand-env`.
> For [convert][], you can use `--extra-args="-enable-features=integrations-next -config.expand-env"`
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 15]
```diff
@@ -297,15 +297,41 @@ loki.write "logs_varlogs" {
...
+
 ## Limitations
 
 Configuration conversion is done on a best-effort basis. {{< param "PRODUCT_ROOT_NAME" >}} will issue warnings or errors where the conversion can't be performed.
 
 After the configuration is converted, review the {{< param "PRODUCT_NAME" >}} configuration file and verify that it's correct before starting to use it in a production environment.
 
-Review the following checklist:
+This following list of limitations is specific to the converter tooling and not {{< param "PRODUCT_NAME" >}}:
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-static.md:332]:
```suggestion
The following list is specific to the convert command and not {{< param "PRODUCT_NAME" >}}:
```

File: docs/sources/flow/getting-started/migrating-from-static.md [hunk 16]
```diff
@@ -297,15 +297,41 @@ loki.write "logs_varlogs" {
...
 
-* The following configuration options are not available for conversion to {{< param "PRODUCT_NAME" >}}: [Integrations next][], [Traces][], and [Agent Management][].
+* The following configuration options are not available for conversion to {{< param "PRODUCT_NAME" >}}: [Traces][] and [Agent Management][].
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-static.md:334]:
```suggestion
* The  [Traces][] and [Agent Management][] configuration options can't be converted to {{< param "PRODUCT_NAME" >}}.
```

File: docs/sources/flow/reference/cli/convert.md [hunk 17]
```diff
@@ -101,6 +103,13 @@ Refer to [Migrate from Promtail to {{< param "PRODUCT_NAME" >}}]({{< relref "../
 Using the `--source-format=static` will convert the source configuration from a
 [Grafana Agent Static]({{< relref "../../../static" >}}) configuration to a {{< param "PRODUCT_NAME" >}} configuration.
 
+Include `--extra-args` for passing additional command line flags from the original format.
+For example, `--extra-args="-enable-features=integrations-next"` will convert a static
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/convert.md:107]:
```suggestion
For example, `--extra-args="-enable-features=integrations-next"` will convert a Grafana Agent Static
```

File: docs/sources/flow/reference/cli/convert.md [hunk 18]
```diff
@@ -101,6 +103,13 @@ Refer to [Migrate from Promtail to {{< param "PRODUCT_NAME" >}}]({{< relref "../
...
+[integrations-next]({{< relref "../../../static/configuration/integrations/integrations-next/" >}})
+configuration to a {{< param "PRODUCT_NAME" >}} configuration. You can also
+expand env vars by including the `--extra-args="-config.expand-env"`. Multiple command line
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/convert.md:110]:
```suggestion
expand environment variables with `--extra-args="-config.expand-env"`. You can combine multiple command line
```

File: docs/sources/flow/reference/cli/convert.md [hunk 19]
```diff
@@ -101,6 +103,13 @@ Refer to [Migrate from Promtail to {{< param "PRODUCT_NAME" >}}]({{< relref "../
...
+flags can be combined with a space between, such as `--extra-args="-enable-features=integrations-next -config.expand-env"`.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/convert.md:111]:
```suggestion
flags with a space between each flag, for example `--extra-args="-enable-features=integrations-next -config.expand-env"`.
```

File: docs/sources/flow/reference/cli/run.md [hunk 20]
```diff
@@ -180,11 +181,15 @@ the source format to River and immediately starts running with the new
...
 
-If you also use the `--config.bypass-conversion-errors` command-line argument,
+If you include the `--config.bypass-conversion-errors` command-line argument,
 {{< param "PRODUCT_NAME" >}} will ignore any errors from the converter. Use this argument
 with caution because the resulting conversion may not be equivalent to the
 original configuration.
 
+Include `--config.extra-args` for passing additional command line flags from
+the original format to the converter. See [grafana-agent-flow convert][]
+for more details on how `extra-args` work.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/run.md:191]:
```suggestion
Include `--config.extra-args` to pass additional command line flags from the original format to the converter.
Refer to [grafana-agent-flow convert][] for more details on how `extra-args` work.
```

**Review by clayton-cornell — APPROVED** [review]

File: docs/sources/flow/getting-started/migrating-from-static.md (hunk 11 above)
Comment (erikbaranowski) [reply docs/sources/flow/getting-started/migrating-from-static.md:302]:
```suggestion
You can convert [integrations next][] configurations by adding the `extra-args` flag for [convert][] or `config.extra-args` for [run][].
```

Comment (erikbaranowski) [reply docs/sources/flow/getting-started/migrating-from-static.md]: I applied something a little different since I think my intended meaning got jumbled with your improved phrasing.

Comment (clayton-cornell) [reply docs/sources/flow/getting-started/migrating-from-static.md]: Looks MUCH better now with the "something a little different" and now it makes sense :-)

File: cmd/internal/flowmode/cmd_convert.go (hunk 7 above)
Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_convert.go]: I'll take a look tomorrow about tidying it up and adding some fail test cases, but what do you think about the new direction for parsing arguments?

**Review by thampiotr — APPROVED** [review]
Nice work, thanks!

File: cmd/internal/flowmode/cmd_convert.go (hunk 7 above)
Comment (thampiotr) [reply cmd/internal/flowmode/cmd_convert.go]: That will do it! thanks :)

File: cmd/internal/flowmode/cmd_convert.go [hunk 21]
```diff
@@ -174,3 +187,35 @@ func supportedFormatsList() string {
 	}
 	return strings.Join(ret, ", ")
 }
+
+func parseExtraArgs(extraArgs string) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_convert.go:191]: once I did some deeper testing, I made some more significant improvements to the func. I added comments throughout due to the complexity which will hopefully help future us
